src/persist-client/src/impl/state.rs

```diff
@@ -122,10 +159,13 @@ where
         &mut self,
         batch: &HollowBatch<T>,
         writer_id: &WriterId,
-    ) -> ControlFlow<Result<Upper<T>, InvalidUsage<T>>, Vec<FueledMergeReq<T>>> {
-        if !self.writers.contains_key(writer_id) {
-            return Break(Err(InvalidUsage::UnknownWriter(writer_id.clone())));
-        }
+    ) -> ControlFlow<Result<Upper<T>, InvalidUsage<T>>, (Vec<FueledMergeReq<T>>, IdempotencyToken)>
+    {
+        let writer = match self.writers.get_mut(writer_id) {
+            None => return Break(Err(InvalidUsage::UnknownWriter(writer_id.clone()))),
+            Some(writer) => writer,
+        };
+        writer.idempotency_token = writer.idempotency_token.next();
```

Comment:
Oh, shoot, it occurs to me that this isn’t right. The idempotency token needs to increase even if the new state fails to apply. Guess I need to store the source of truth somewhere else. Hmm.